collection.go

```diff
@@ -575,6 +536,7 @@ func (cl *collectionLoader) populateMaps() error {
 					if err != nil {
 						return fmt.Errorf("loading program %s, for map %s: %w", objName, mapName, err)
 					}
+					mapSpec = mapSpec.Copy()
```

Comment:
This will copy the outer map for every item in `mapSpec.Contents`, which is not great.

An `if len(mapSpec.Contents) != 0 { }` around the existing `mapSpec = mapSpec.Copy()` above should do the trick.

Nice find, otherwise!

Reply:
The problem with that is that it copies `.rodata` maps as well, which it would be nice to avoid!

Reply:
Solved it a little bit differently, should  be ok now hopefully.

Reply:
I pushed a fixup that uses `canStoreMap()` and `canStoreProgram()` for everything here and removes a level of indentation.